diff --git a/index.js b/index.js
index f3584268408a88cc1a8380a19b5bcbdf38579e23..7ed7a076e13514a8455160e98add8a8093147196 100644
--- a/index.js
+++ b/index.js
@@ -128,7 +128,7 @@ OffsetBuffer.prototype.copy = function copy(target, targetOff, off, n) {
 
 OffsetBuffer.prototype.take = function take(n) {
   if (n === 0)
-    return new Buffer(0);
+    return Buffer.alloc(0);
 
   this.size -= n;
 
@@ -148,7 +148,7 @@ OffsetBuffer.prototype.take = function take(n) {
   }
 
   // Allocate and fill buffer
-  var out = new Buffer(n);
+  var out = Buffer.allocUnsafe(n);
   var toOff = 0;
   var startOff = this.offset;
   for (var i = 0; toOff !== n && i < this.buffers.length; i++) {
